src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/StripJavaDebugAttributesPlugin.java

```diff
@@ -59,10 +59,8 @@ public ResourcePool transform(ResourcePool in, ResourcePoolBuilder out) {
                     if (path.endsWith("module-info.class")) {
                         // XXX. Do we have debug info? Is Asm ready for module-info?
                     } else {
-                        ClassReader reader = newClassReader(path, resource);
-                        ClassWriter writer = new ClassWriter(ClassWriter.COMPUTE_MAXS);
-                        reader.accept(writer, ClassReader.SKIP_DEBUG);
-                        byte[] content = writer.toByteArray();
+                        byte[] content = newClassReader(path, resource, Classfile.Option.processDebug(false))
```

Comment:
ASM `ClassReader.SKIP_DEBUG` flag indicates to skip  the SourceFile, SourceDebugExtension, LocalVariableTable, LocalVariableTypeTable, LineNumberTable and MethodParameters attributes.

`Classfile.Option.processDebug(false)` only skips LocalVariableTable, LocalVariableTypeTable, CHARACTER_RANGE_TABLE attributes.   To implement the same behavior as `SKIP_DEBUG`, it needs `Classfile.Option.processLineNumbers(false)` and skipping other attributes.

Reply:
Right, I'll fix it, thanks.

Reply:
Looks like `LineNumberTable` is not stripped even with `Option.processLineNumbers(false)`.

Reply:
Yes, I forgot to transform code elements, where  `Classfile.Option.processLineNumbers(false)` option is applied.
Thanks for catching it.

We may also consider a new form of transformation allowing to filter code attributes without inflation of all code elements (in the Classfile API).